dpnp/dparray.pyx

```diff
@@ -77,9 +156,10 @@ cdef class dparray:
 
     """
 
-    def __init__(self, shape, dtype=float64, memptr=None, strides=None, order=b'C'):
+    def __init__(self, shape, dtype=float64, memory=None, strides=None, order=b'C'):
```

Comment:
it looks like it is not the best name for this parameter. This is not a memory or memory pointer.
It looks like it is `external object`
Do you have any examples in other API where such functionality used? Perhaps it is better to be aligned with names,

Reply:
In NumPy, the keyword is call `buffer=`. The buffer needs to be a byte-like object. NumPy would not allow an arbitrary object to be used in its place even if that object implements `__array_interface__`. 

NumPy can consume duck arrays that advertise themselves as arrays only via `__array_interface__` property only via `numpy.asarray`, or `numpy.asanyarray`.